frontend/public/components/storage/attach-storage.tsx

```diff
@@ -0,0 +1,468 @@
+      const storage = {
+        attach: null,
+        existingMountPaths: null,
+        volumeAlreadyMounted: null,
+        onPVCSelected: null,
+      };
+      storage.attach = {
+        persistentVolumeClaim: claimName,
+        volumeName: volumeName,
+        mountPath: mountPath,
+        subPath: subPath,
+        readOnly: this.state.readOnly,
+        allContainers: true,
+        containers: {},
+      };
+      const isContainerSelected = function(container) {
+        return (
+          storage.attach.allContainers ||
+          storage.attach.containers[container.name]
+        );
+      };
+
+      //   Look at the existing mount paths so that we can warn if the new value is not unique.
+      storage.existingMountPaths = getMountPaths(
+        _.get(resourceObj.spec, 'template'),
+        isContainerSelected
+      );
+      const updateVolumeName = function() {
+        const pvc = _.get(resourceObj.spec, 'volumes.persistentVolumeClaim');
+        if (!pvc) {
+          return;
+        }
+
+        // Check if there is already a volume for this PVC.
+        let volumes = _.get(resourceObj.spec, 'template.spec.volumes');
+        let volume = _.find(volumes, {
+          persistentVolumeClaim: {
+            claimName: pvc.metadata.name,
+          },
+        });
+        if (volume) {
+          // If there's already a  volume, reuse the volume name.
+          storage.attach.volumeName = resourceObj.spec.volumes.metadata.name;
+          storage.volumeAlreadyMounted = true;
+        } else if (storage.volumeAlreadyMounted) {
+          // Clear the volume name value since it was associated with the previously selected PVC.
+          storage.attach.volumeName = '';
+          storage.volumeAlreadyMounted = false;
+        }
+      };
+      storage.onPVCSelected = updateVolumeName;
+
+      // generate a volume name if not provided
+      if (!volumeName) {
+        // Create a volume name using the project
+        volumeName = `volume-${metadata.namespace}`;
+      }
+      if (mountPath) {
+        // for each container in the pod spec, add the new volume mount
+
+        _.each(template.spec.containers, function(container) {
+          if (isContainerSelected(container)) {
+            let newVolumeMount = createVolumeMount(
+              volumeName,
+              mountPath,
+              subPath,
+              readOnly
+            );
+            if (!container.volumeMounts) {
+              container.volumeMounts = [];
+            }
+            container.volumeMounts.push(newVolumeMount);
+          }
+        });
+      }
+
+      // add the new volume to the pod template
+      if (!storage.volumeAlreadyMounted) {
+        template.spec.volumes = template.spec.volumes || [];
+        template.spec.volumes.push(
+          createVolume(volumeName, storage.attach.persistentVolumeClaim)
+        );
+      }
+
+      this.setState({ resourceObj, inProgress: true });
```

Comment:
I'm not sure why we should set `resourceObj` back to state. Won't this leave things there that we don't want if I try to submit the form multiple times, changing some field values?